In [14]:
import pandas as pd
import statistics as st

# 1. Find out how many males and females participated in the test.

In [41]:
data = pd.read_csv('StudentsPerformance.csv')
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [10]:
data.groupby('gender')['gender'].count()

gender
female    518
male      482
Name: gender, dtype: int64

From this its understood that the totl no.of males participated is 482 and females 518 and thus more no.of females particiapted the test

# 2. What do you think about the students' parental level of education?

In [12]:
data.groupby('parental level of education')['parental level of education'].count()

parental level of education
associate's degree    222
bachelor's degree     118
high school           196
master's degree        59
some college          226
some high school      179
Name: parental level of education, dtype: int64

In [16]:
st.mode(data['parental level of education'])

'some college'

We took the mode here and from this ts clear that most no.of parents has gone to some colleges compared to other categories

# 3. Who scores the most on average for math, reading and writing based on


# ● Gender

In [21]:
data.groupby('gender')['math score'].mean()

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64

In [24]:
df = data.groupby('gender')['math score','reading score','writing score'].mean()
df.idxmax()

<ipython-input-24-6f60ead8b0c7>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df = data.groupby('gender')['math score','reading score','writing score'].mean()


math score         male
reading score    female
writing score    female
dtype: object

Based on gender,males score most on average for math and females score most on average for reading and writing score

# ● Test preparation course

In [31]:
df1 = data.groupby('test preparation course')['math score','reading score','writing score'].mean()
df1

<ipython-input-31-b139891a3310>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df1 = data.groupby('test preparation course')['math score','reading score','writing score'].mean()


math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

In [42]:
df1.idxmax()

math score       completed
reading score    completed
writing score    completed
dtype: object

Above result indicates that the students who have completed the test preparation has scored the most on math,reading and writing

# What do you think about the scoring variation for math, reading and writing


# ● Gender

In [36]:
df3 = data.groupby('gender')['math score','reading score','writing score'].std()
df3

<ipython-input-36-64137c33012b>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df3 = data.groupby('gender')['math score','reading score','writing score'].std()


math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

Scoring variations for math,reading and writing are as listed above.Thus females have more variation compared to males

# ● Test preparation course


In [43]:
df4 = data.groupby('test preparation course')['math score','reading score','writing score'].std()
df4

<ipython-input-43-c89aa2d3199d>:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df4 = data.groupby('test preparation course')['math score','reading score','writing score'].std()


math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

Scoring varaitions based on test preparation course -completed and none is as listed above for each scores.Thus 'none' category has more variation

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.


In [40]:
data['math score'].nlargest(250)

515    100
516    100
517    100
996    100
997    100
      ... 
428     77
429     77
849     77
850     77
851     77
Name: math score, Length: 250, dtype: int64

Above students with listed IDs come in top 25% of students based on maths score and thus they can be given bonus points